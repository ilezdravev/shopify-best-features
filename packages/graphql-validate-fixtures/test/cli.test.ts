import {resolve} from 'path';
import {execSync} from 'child_process';

const scriptPath = resolve(__dirname, '../bin/graphql-validate-fixtures');
const rootFixturePath = resolve(__dirname, 'fixtures');

describe('cli', () => {
  it('succeeds when there are no fixture errors', () => {
    expect(() => execSync(cliCommandForFixtureDirectory('all-clear'))).not.toThrowError();
  });

  it('fails when there are fixture errors', () => {
    expect(() => execSync(cliCommandForFixtureDirectory('fixture-errors'))).toThrowErrorMatchingSnapshot();
  });

  it('fails when there are syntax errors', () => {
    expect(() => execSync(cliCommandForFixtureDirectory('fixture-invalid'))).toThrowErrorMatchingSnapshot();
    expect(() => execSync(cliCommandForFixtureDirectory('malformed-query'))).toThrowErrorMatchingSnapshot();
    expect(() => execSync(cliCommandForFixtureDirectory('missing-schema'))).toThrowErrorMatchingSnapshot();
  });
});

function cliCommandForFixtureDirectory(fixture: string) {
  const fixtureDirectory = resolve(rootFixturePath, fixture);
  return [
    scriptPath,
    `'${resolve(fixtureDirectory, 'fixtures/**/*.json')}'`,
    `--schema-path '${resolve(fixtureDirectory, 'schema.json')}'`,
    `--operation-paths '${resolve(fixtureDirectory, 'queries/**/*.graphql')}'`,
  ].join(' ');
}
