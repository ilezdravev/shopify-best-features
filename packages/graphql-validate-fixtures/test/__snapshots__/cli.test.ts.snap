// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli fails when there are fixture errors 1`] = `
"Command failed: /Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/bin/graphql-validate-fixtures '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/fixture-errors/fixtures/**/*.json' --schema-path '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/fixture-errors/schema.json' --operation-paths '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/fixture-errors/queries/**/*.graphql'
"
`;

exports[`cli fails when there are syntax errors 1`] = `
"Command failed: /Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/bin/graphql-validate-fixtures '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/fixture-invalid/fixtures/**/*.json' --schema-path '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/fixture-invalid/schema.json' --operation-paths '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/fixture-invalid/queries/**/*.graphql'
"
`;

exports[`cli fails when there are syntax errors 2`] = `
"Command failed: /Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/bin/graphql-validate-fixtures '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/malformed-query/fixtures/**/*.json' --schema-path '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/malformed-query/schema.json' --operation-paths '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/malformed-query/queries/**/*.graphql'
"
`;

exports[`cli fails when there are syntax errors 3`] = `
"Command failed: /Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/bin/graphql-validate-fixtures '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/missing-schema/fixtures/**/*.json' --schema-path '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/missing-schema/schema.json' --operation-paths '/Users/chrissauve/Projects/graphql-tools/packages/graphql-validate-fixtures/test/fixtures/missing-schema/queries/**/*.graphql'
"
`;
